hpx/runtime/applier/apply_helper.hpp

```diff
@@ -116,39 +115,78 @@ namespace hpx { namespace applier { namespace detail
         }
     };
 
+    ///////////////////////////////////////////////////////////////////////////
+    HPX_FORCEINLINE bool needs_to_execute_asynchronously()
+    {
+        bool async_work = false;
+
+#if defined(HPX_HAVE_THREADS_GET_STACK_POINTER)
+        std::ptrdiff_t remaining_stack = this_thread::get_available_stack_space();
+
+        if(remaining_stack < 0)
+        {
+            HPX_THROW_EXCEPTION(out_of_memory,
+                "hpx::applier::detail::needs_to_execute_asynchronously",
+                "Stack overflow");
+        }
+
+        async_work = remaining_stack < (8 * HPX_THREADS_STACK_OVERHEAD);
+#endif
```

Comment:
You are right ... the continuation recursion count might not make sense at all in that context.